azurerm/internal/services/machinelearning/machine_learning_inference_cluster_resource.go

```diff
@@ -0,0 +1,498 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+const min_number_of_nodes_prod int32 = 12
+
+func resourceAksInferenceCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAksInferenceClusterCreateUpdate,
+		Read:   resourceAksInferenceClusterRead,
+		Update: resourceAksInferenceClusterCreateUpdate,
+		Delete: resourceAksInferenceClusterDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.InferenceClusterID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.InferenceClusterName,
+			},
+
+			"workspace_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.WorkspaceName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"kubernetes_cluster_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"kubernetes_cluster_rg": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.KubernetesClusterResourceGroupName,
+			},
+
+			"cluster_purpose": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "Dev",
+				ValidateFunc: validate.ClusterPurpose,
+			},
+
+			"node_pool_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NodePoolName,
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(machinelearningservices.SystemAssigned),
+							}, false),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"ssl_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+
+			"ssl_certificate_custom": {
```

Comment:
i think we can remove custom as its implied?
```suggestion
			"ssl_certificate": {
```

Reply:
I was thinking about doing a `ssl_certificate_custom` and `ssl_certificate_microsoft` block, but the Microsoft SSL configuration does not seem to work (just specifying `leaf_domain_label` still requests cert and key files). In the second commit I just go for `ssl` here and I removed the `ssl_enabled` parameter as I think this can be implied if you specify non-empty certificates and keys.

Reply:
ssl_certificate/ssl_certificate_microsoft would work

Reply:
Does that mean you are not happy with:
```bash
ssl {
      cert = file("testdata/cert.pem")
 	 key = file("testdata/key.pem")
 	 cname = "www.contoso.com"
   }
```
In principle the schema could be - correct me if I am wrong - improved in future (if the Microsoft bug does not occur - I think it is a bug) as follows:
```bash
			"ssl": {
				Type:     schema.TypeSet,
				Optional: true,
				ForceNew: true,
				MaxItems: 1,
				Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						"cert": {
							Type:     schema.TypeString,
							Optional: true,
							Default:  "",
                                                        ConflictsWith: {"leaf_domain_label", "overwrite_existing_domain"},
						},
						"key": {
							Type:     schema.TypeString,
							Optional: true,
							Default:  "",
                                                        ConflictsWith: {"leaf_domain_label", "overwrite_existing_domain"},
						},
						"cname": {
							Type:     schema.TypeString,
							Optional: true,
							Default:  "",
                                                        ConflictsWith: {"leaf_domain_label", "overwrite_existing_domain"},
						},
                                               	"leaf_domain_label": {
							Type:     schema.TypeString,
							Optional: true,
							Default:  "",
                                                        ConflictsWith: {"cert", "key", "cname"},
						},
                                               	"overwrite_existing_domain": {
							Type:     schema.TypeString,
							Optional: true,
							Default:  "",
                                                        ConflictsWith: {"cert", "key", "cname"},
						},
					},
				},
			},
```
What do you think is best for possible future enhancements: 2 separate blocks with now only an "ssl_certificate" block or everything in the "ssl" block or if you want "ssl_certificate" block?